src/Compilers/VisualBasic/Portable/Binding/Binder_Invocation.vb

```diff
@@ -2253,8 +2249,7 @@ ProduceBoundNode:
                             ReportDiagnostic(diagnostics, diagnosticLocation, If(queryMode, ERRID.ERR_TypeInferenceFailureNoExplicitNoBest2, ERRID.ERR_TypeInferenceFailureNoBest2), If(representCandidateInDiagnosticsOpt, candidateSymbol))
                         End If
                     Else
-                        If candidateAnalysisResult.TypeArgumentInferenceDiagnosticsOpt IsNot Nothing AndAlso
-                           candidateAnalysisResult.TypeArgumentInferenceDiagnosticsOpt.HasAnyResolvedErrors Then
+                        If candidateAnalysisResult.TypeArgumentInferenceDiagnosticsOpt.HasAnyResolvedErrors Then
```

Comment:
I didn't understand why the null test was removed. Couldn't `TypeArgumentInferenceDiagnosticsOpt` still be null?

Reply:
> I didn't understand why the null test was removed. Couldn't TypeArgumentInferenceDiagnosticsOpt still be null?

It is a ```ImmutableBindingDiagnostic(Of AssemblySymbol)``` now, which is a struct.